File: grade1.js
// 10172번
console.log(`|\\_/|
|q p|   /}
( 0 )"""\\
|"^"\`    |
||_/=\\\\__|
`)

//10869번
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split(' ');
console.log(input.reduce((prev, curr) => Number(prev) + Number(curr)))
console.log(input.reduce((prev, curr) => Number(prev) - Number(curr)))
console.log(input.reduce((prev, curr) => Number(prev) * Number(curr)))
console.log(input.reduce((prev, curr) => Math.floor(Number(prev) / Number(curr))))
console.log(input.reduce((prev, curr) => Number(prev) % Number(curr)))

//10430번
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split(' ');

for (i = 0; i < input.length; i++){
    input[i] = Number(input[i]);
}
console.log( ( input[0] + input[1] ) % input[2] );
console.log( ((input[0] % input[2]) + (input[1] % input[2])) % input[2] );
console.log( (input[0] * input[1]) % input[2] );
console.log( ((input[0] % input[2]) * (input[1] % input[2])) % input[2] );

// 2588번 
// 1) 배열 index 활용 (M 7076KB, T 108ms, C 361B)
const fs = require('fs');
const arr = fs.readFileSync('/dev/stdin').toString().split('\n');
const b_arr = arr[1].split("");
for (i = 0; i < b_arr.length; i++){
    arr[i] = Number(arr[i]);
    b_arr[i] = Number(b_arr[i]);
}
arr.pop();
console.log(arr[0] * b_arr[2]);
console.log(arr[0] * b_arr[1]);
console.log(arr[0] * b_arr[0]);
console.log(arr[0] * arr[1]);

// 2) 배열 값 사칙연산 (M 7080KB, T 104ms, C 306B)
const fs = require('fs');
const a = fs.readFileSync('/dev/stdin').toString().split('\n');

for (i = 0; i < a.length; i++){
    a[i] = Number(a[i])
}
console.log(a[0] * ( a[1] % 10 ));
console.log(a[0] * ( parseInt( a[1] / 10 ) % 10 ));
console.log(a[0] * parseInt( a[1] / 100 ));
console.log(a[0] * a[1]);

File: grade1.py
#10172번
print("""\
|\_/|
|q p|   /}
( 0 )\"\"\"\\
|"^"`    |
||_/=\\\\__|\
""")

#1000번
#1) map 함수 사용 (M 29380KB, T 64ms, C 59B)
a = list(map(int, input("").split(" ")))
print(a[0] + a[1])

#2) 리스트 내포(List comprehension) (M 29380KB, T 64ms, C 83B)
a = input("").split(" ")
list_a = [int (i) for i in a]
print(list_a[0] + list_a[1])

#10869번
a = list(map(int, input("").split(" ")))
print(a[0] + a[1])
print(a[0] - a[1])
print(a[0] * a[1])
print(int(a[0] / a[1]))
print(a[0] % a[1])

#10430번
a = list(map(int, input("").split(" ")))

print( (a[0] + a[1] ) % a[2] )
print( ((a[0] % a[2]) + (a[1] % a[2])) % a[2] )
print( (a[0] * a[1]) % a[2] )
print( ((a[0] % a[2]) * (a[1] % a[2])) % a[2] )

#2588번
a, b = int(input("")), int(input(""))
print(a * (b % 10))
print(a * ((b % 100) // 10))
print(a * (b // 100))
print(a * b)